### ISSUE TITLE
Configuring Xcode project settings for Slather

### ISSUE THREAD

**Issue by smileyborg** [venmo/slather#73]
Running the recommended

    $ slather setup path/to/project.xcodeproj

ends up executing [this code](https://github.com/venmo/slather/blob/6a1ad3462f1f44287d3fecc34a75061a6e04bf98/lib/slather/project.rb#L9-L14):

```ruby
    def slather_setup_for_coverage
      build_configurations.each do |build_configuration|
        build_configuration.build_settings["GCC_INSTRUMENT_PROGRAM_FLOW_ARCS"] = "YES"
        build_configuration.build_settings["GCC_GENERATE_TEST_COVERAGE_FILES"] = "YES"
      end
    end
```

As currently written, I think that the effects of this (or manually modifying the project file in the same way) is problematic, because we end up turning these 2 flags on even for non-Debug/Test/Coverage builds. From my understanding of these flags, changing them from their default `NO` value ends up changing the compiled binary (as additional code is injected to perform the measurements). At the very least, this is bad because it's likely inefficient and causes unnecessary overhead. But it's also risky for production code, and a quick search on Google turns up some reports of app crashes and other nastiness in some situations when these flags are enabled.

Now, I don't think this will affect users who integrate the library with CocoaPods, since CocoaPods is generating its own Xcode project (right?), but if you use Carthage or a similar mechanism to build a framework or static library directly from the Xcode project in the repo itself, having these flags on for all build configurations will affect the generated artifacts.

I think these flags should only be set for Debug/Test/Coverage build configurations. The reason I am opening this issue instead of a PR is because I'm not sure the best way to do this...we could test against the build configuration name (looking for "Debug" or similar), or check for a preprocessor define of "DEBUG", or potentially something else.

Thoughts?

**Comment by marklarr** [comment]
If having those bugs on can lead to crashes in a compiled binary, then Apple's recommendation of using a separate build configuration for coverage builds seems like the best option.

Maybe Slather can determine which configuration you are using for testing by looking at your scheme's test configuration

![monosnap 2015-05-06 22-41-28](https://cloud.githubusercontent.com/assets/760261/7509101/1a243412-f441-11e4-8df5-2500db6b7f3b.png)

and then it can duplicate this configuration into a new `Test` configuration, set it up for coverage, and then change your scheme to use that as the build configuration. And of course, give some output so the user knows that we just mucked with things.

As for the Pods projects, Slather can reliably assume that Cocoapods will set them up with a `Release` and `Debug` configuration, and just make a new `Test` configuration from `Debug`, set it up for coverage,  etc.

Does that sound reasonable? This is a good find.

**Comment by smileyborg** [comment]
Interesting suggestion, but it seems pretty heavy-handed to me. I think it would be worth soliciting opinions on this from more folks before trying to make a decision.

**Comment by marklarr** [comment]
The other option is asking people which configuration they would like to be copied over into a new one.

Going forward, I want Slather to continue to remove the click-work for the end-user, meaning that I don't want people to be required to go into Xcode and click through the project settings just to use Slather. And, at any rate, we will need some sort of automated solution for the `Pods` project, because not everybody checks those in, and setting them up manually every time they are created is not a reasonable expectation.

**Comment by marklarr** [comment]
@kylef @ayanonagon

**Comment by marklarr** [comment]
@mickeyreiss

**Comment by Ashton-W** [comment]
I personally use slather as part of a *test and show coverage* task.
Running: 
```bash
xcodebuild test ... GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES
```
instead of making any changes to the project.

**Comment by tarbrain** [comment]
In fact that would be the cleanest way to do it. It keeps slather completely outside the Podfile and all involved projects.

**Comment by tarbrain** [comment]
I have tried it out on https://github.com/tarbrain/TBStateMachine:

- removed the slather plugin from the `Podfile`
- removed `GCC_INSTRUMENT_PROGRAM_FLOW_ARCS` and `GCC_GENERATE_TEST_COVERAGE_FILES` flags from the project
- only adding flags in the build script

Works good.

**Comment by marklarr** [comment]
thanks @tarbrain for validating, I'll look into this when I have the chance, or if somebody else wants to that'd be great too :)

**Comment by neonichu** [comment]
I don't think it makes sense to spend time on actually implementing a better setup experience at this point, considering this became a simple checkbox in Xcode 7. My plan would be removing the setup option entirely and documenting how to do this manually for legacy projects still using Xcode 6.
